gradle-plugin/src/proguard/gradle/configuration/ProGuardTaskConfiguration.java

```diff
@@ -0,0 +1,473 @@
+ *
+ * For clarity, fields on `Configuration` that are not task inputs or outputs are exposed with `@Internal`.
+ * These methods could just as easily be removed from this type.
+ */
+public class ProGuardTaskConfiguration {
+    protected final Provider<Configuration> configurationProvider;
+    private final ObjectFactory objectFactory;
+
+    public ProGuardTaskConfiguration(Provider<Configuration> configurationProvider, ObjectFactory objectFactory) {
+        this.configurationProvider = configurationProvider;
+        this.objectFactory = objectFactory;
+    }
+
+    public static ProGuardTaskConfiguration create(Provider<Configuration> configurationProvider, ObjectFactory objectFactory) {
+        if (GradleVersionCheck.isAtLeastGradle7()) {
+            return new ProGuardTaskConfiguration(configurationProvider, objectFactory);
+        } else {
+            return new ProGuardTaskConfigurationForGradle6(configurationProvider, objectFactory);
+        }
+    }
+
+    private Configuration getConfiguration() {
+        return configurationProvider.get();
+    }
+
+    protected <T> Provider<T> mappedProvider(Transformer<T, Configuration> transformer) {
+        return configurationProvider.map(transformer);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Input and output options.
+    ///////////////////////////////////////////////////////////////////////////
+
+    @Classpath
+    public FileCollection getInputJars() {
+        Provider<List<File>> inputJarsProvider = mappedProvider(config -> {
+            List<File> files = new ArrayList<>();
+            ClassPath programJars = config.programJars;
+            for (int i = 0; i < programJars.size(); i++) {
+                ClassPathEntry classPathEntry = programJars.get(i);
+                if (!classPathEntry.isOutput()) {
+                    files.add(classPathEntry.getFile());
+                }
+            }
+
+            ClassPath libraryJars = config.libraryJars;
+            for (int i = 0; i < libraryJars.size(); i++) {
+                ClassPathEntry classPathEntry = libraryJars.get(i);
+                if (!classPathEntry.isOutput()) {
+                    files.add(classPathEntry.getFile());
+                }
+            }
+            return files;
+        });
+        return objectFactory.fileCollection().from(inputJarsProvider);
+    }
+
+    @OutputFiles
+    public FileCollection getOutputFiles() {
+        Configuration config = getConfiguration();
+
+        ConfigurableFileCollection files = objectFactory.fileCollection();
+        ClassPath programJars = config.programJars;
+        for (int i = 0; i < programJars.size(); i++) {
+            ClassPathEntry classPathEntry = programJars.get(i);
+            if (classPathEntry.isOutput() && !classPathEntry.isDirectory()) {
+                files.from(classPathEntry.getFile());
+            }
+        }
+        return files;
```

Comment:
I think you really should create the file collection in the constructor and then augment it in this method, or add it via a callable.

Reply:
Fixed